projects/angular/src/data/datagrid/render/main-renderer.ts

```diff
@@ -142,14 +142,13 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, Af
     if (this.headers) {
       if (state && !this.columnsService.hasCache()) {
         this.columnsService.cache();
-        this.headers.forEach((_header, index) => {
-          if (index > 0) {
-            this.columnsService.emitStateChangeAt(index, {
-              changes: [DatagridColumnChanges.HIDDEN],
-              hidden: state,
-            });
-          }
-        });
+        const firstVisibleColumnIndex = this.columnsService.columnStates.findIndex(state => !state.hidden);
+        for (let index = firstVisibleColumnIndex + 1; index < this.headers.length; index++) {
+          this.columnsService.emitStateChangeAt(index, {
+            changes: [DatagridColumnChanges.HIDDEN],
+            hidden: state,
+          });
+        }
```

Comment:
```suggestion
        this.columnsService.visibleColumns.forEach((header, index) => {
         if (index > 0) {
          this.columnsService.emitStateChangeAt(header.columnIndex, {
            changes: [DatagridColumnChanges.HIDDEN],
            hidden: state,
          });
         }
```

Reply:
needed to add a missing `});` at the end so I pushed another commit instead of applying the suggestion from the web